graylog2-server/src/main/java/org/graylog2/shared/plugins/ChainingClassLoader.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.shared.plugins;
+
+import com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class ChainingClassLoader extends ClassLoader {
+    public static final Logger LOG = LoggerFactory.getLogger(ChainingClassLoader.class);
+
+    private final List<ClassLoader> classLoaders = new CopyOnWriteArrayList<>();
+
+    public ChainingClassLoader(ClassLoader parent) {
+        classLoaders.add(parent);
```

Comment:
I've opted to use this variant (with the parent class loader being the first item in the class loaders list) to simplify the overridden methods. Otherwise they'd need two code paths: One calling the corresponding method in `super` and one iterating over the list of class loaders.